### PR TITLE
Use vue-test-utils for testing

### PR DESCRIPTION
**Pull request by josephting**
## Description

1. Added `vue-test-utils` to package.json
2. Rewritten all tests to use `vue-test-utils` retaining the original method used in each of the tests.

I have noticed some tests aren't doing anything while other does not generate exact same snapshot but those fixes will come later.

## Motivation and Context

Effort being placed in so that work on increasing code coverage can eventually be carried out for the project.

## How Has This Been Tested?

1. All the assertion have been negated i.e. adding `.not` or removing `.not` to test if they can fail
2. `precommit` passed as before

## Screenshots (if appropriate):

N/A

## Types of changes

- [ ] ~Bug fix (non-breaking change which fixes an issue)~
- [x] New feature (non-breaking change which adds functionality)
- [ ] ~Breaking change (fix or feature that would cause existing functionality to not work as expected)~

## Checklist:

- [x] My code follows the code style of this project.
- [ ] ~My change requires a change to the documentation.~
  - [ ] ~I have updated the documentation accordingly.~
- [ ] ~I have included a vue-play example (if this is a new feature)~

## Todo:

- [ ] Fix `should apply 200px carousel width when element has 200px width` test case (currently, it does not fail even after adding `.not` to the assertion)

### CODE REVIEW

File: tests/client/components/__snapshots__/carousel.spec.js.snap [hunk 1]
```diff
@@ -258,7 +258,7 @@ exports[`Carousel should reset autoplay when switching slide without autoplayHov
 
 exports[`Carousel should set carousel height to slide height 1`] = `
 ".VueCarousel-wrapper
-  .VueCarousel-inner(role='listbox', style='transform: translate(0px, 0); visibility: hidden; height: 200px;')
+  .VueCarousel-inner(role='listbox', style='transform: translate(0px, 0); visibility: hidden; height: auto;')
```
Comment (josephting) [inline tests/client/components/__snapshots__/carousel.spec.js.snap]: With vue-test-utils, the html doesn't seems to get updated even after `$nextTick`.
If anyone knows what's up with this, your input is very much welcomed.

Comment (quinnlangille) [reply tests/client/components/__snapshots__/carousel.spec.js.snap]: @ashleysimpson any idea about this? I've encountered similar stuff in our internal tests but we're working around it in the same manner. Not sure if there is a better alternative

**Comment by quinnlangille** [comment]
Great work @josephting. The PR review form really show us how little tests we have, so super glad you're taking on this initiative. I'll wait for a response from @ashleysimpson on your comment re: snapshots, but otherwise looks good :octocat:

File: tests/client/components/__snapshots__/carousel.spec.js.snap (hunk 1 above)
Comment (ashleysimpson) [reply tests/client/components/__snapshots__/carousel.spec.js.snap]: @josephting after a little bit of research, if you add the property `sync: false` when you mount the wrapper the html will update as expected

Comment (josephting) [reply tests/client/components/__snapshots__/carousel.spec.js.snap]: @ashleysimpson That was a huge help. Thank you so much!
@quinnlangille With that, this PR will only have changes to the test scripts and the snapshots will be untouched.

**Comment by josephting** [comment]
Thank you, @quinnlangille. This PR is ready to be merged now. Then, we can have #337 merged as well.

**Comment by quinnlangille** [comment]
Great! Will merge into the working branch now, really great work @josephting 🚀

**Pull request merged**